src/steps/findings.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  createDirectRelationship,
+  createIntegrationEntity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+  IntegrationMissingKeyError,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../client';
+import { IntegrationConfig } from '../types';
+import { getVulnerabilityLink, getVulnerabilityNumber } from '../util';
+
+export async function fetchFindings({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await apiClient.iterateFindings(async (finding) => {
+    const findingProps = finding.resource;
+    const findingEntity = await jobState.addEntity(
+      createIntegrationEntity({
+        entityData: {
+          source: finding,
+          assign: {
+            _type: 'cobalt_finding',
+            _class: 'Finding',
+            _key: findingProps.id,
+            tag: findingProps.tag,
+            name: findingProps.title,
+            displayName: findingProps.title,
+            description: findingProps.description,
+            category: 'Penetration Testing',
+            typeCategory: findingProps.type_category,
+            labels: JSON.stringify(findingProps.labels, null, 2),
+            impact: JSON.stringify(findingProps.impact, null, 2), //required to be a string in J1 Finding
+            severity: JSON.stringify(findingProps.impact, null, 2), //required property in J1 Finding
+            numericSeverity: findingProps.impact, //required property in J1 Finding
```

Comment:
What are the possible values here? Ideally, `numericSeverity` will be normalized across `Finding` entities in J1. We need better documentation of that fact in the data-model schema.

Reply:
@aiwilliams  It's a range of 1 to 5. Does that match J1's range?